src/bundler/mod.rs

```diff
@@ -130,15 +131,29 @@ impl Bundler {
             )
             .tx
             .clone();
-        tx.set_nonce(nonce).set_chain_id(self.chain_id.as_u64());
+        let gas = client.clone().estimate_gas(&tx, None).await?;
+        tx.set_gas(gas)
+            .set_nonce(nonce)
+            .set_chain_id(self.chain_id.as_u64());
 
+        trace!("Prepare the transaction {tx:?} send to execution client!");
         let tx = client.send_transaction(tx, None).await?;
         let tx_hash = tx.tx_hash();
-        debug!("Send bundle with transaction: {tx:?}");
+        trace!("Send bundle with transaction: {tx:?}");
 
-        // TODO: check if bundle was included on-chain
-        // let res = tx.await?;
-        // debug!("Send bundle with receipt: {res:?}");
+        info!("Send handlePastEvents request");
```

Comment:
I am not sure if we need to wait for the event here.

The bundler should automatically listen for past events when started - and do it all the time.

Reply:
It doesn't wait here. It would just simply check the history events. It is possible that the bundles just sent are not included in blocks at this event handling.

Reply:
Ok, my bad, then it makes sense.

But we should have `handle_past_events` also for bundles submitted by other bundlers (I guess we can call handle past events after each new block is added - and handle reorgs also in the future).

Reply:
We can create issue to handle these two things later.